dali/pipeline/data/tensor_vector.h

```diff
@@ -123,6 +123,12 @@ class DLL_PUBLIC TensorVector {
 
   DLL_PUBLIC void Resize(const TensorListShape<> &new_shape, const TypeInfo &new_type);
 
+  /**
+   * Change the number of tensors in TensorVector.
+   * @param new_size
+   */
+  void SetSize(int new_size);
```

Comment:
I wouldn't change type to unsigned, but I would suggest some naming that differs more from Resize.
Maybe SetBatchSize?

Reply:
Also, it needs more docs, and I already don't like the semantics. Where do you intend to use it? Why does resizing the batch size repeat the last shape? I'd say it's weird.